dvc/ignore.py

```diff
@@ -191,8 +193,15 @@ def __init__(self, fs, root_dir):
             ignore_subrepos=True,
         )
 
-    def _update_trie(self, dirname: str, trie: PathStringTrie) -> None:
-        old_pattern = trie.longest_prefix(dirname).value
+    def _get_key(self, path):
+        parts = self.fs.path.relparts(path, self.root_dir)
+        if parts == (".",):
+            parts = ()
+        return parts
```

Comment:
Obviously need to put into a common utility for repofs/dvcfs/gitfs/dvcignore/etc. Will be introduced separately.